tests/unit/s2n_cert_validation_callback_test.c

```diff
@@ -222,7 +222,7 @@ int main(int argc, char *argv[])
             DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
             EXPECT_NOT_NULL(conn);
             EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
-            EXPECT_SUCCESS(s2n_set_server_name(conn, "s2nTestServer"));
+            EXPECT_SUCCESS(s2n_set_server_name(conn, "localhost"));
```

Comment:
Changing the certificate common name of the default certificate is someone risky since there could be poorly written tests that expect a failure, and now start failing due to a mismatch in the server name rather than what they were trying to test. Before making this change, could we audit the tests that set `s2nTestServer` as the server name and see if any of them expect a generic failure (or a failure that overlaps with the mismatch error)?

This test catches this, both because it checks a success case and looks for a specific error in the failure cases. But some other tests might not be doing either of these things.

Reply:
I searched for tests that sets the server name as `s2nTestServer` and all of them are using the old cert `S2N_RSA_2048_PKCS1_LEAF_CERT` so this should be fine